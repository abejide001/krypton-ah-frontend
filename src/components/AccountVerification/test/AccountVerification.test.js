import React from 'react';
import { shallow } from 'enzyme';
import { Verification } from '..';

describe('AccountVerification Component', () => {
  let wrapper;
  let props;
  let spy;
  beforeAll(() => {
    props = {
      isAuthenticated: true,
      activate: jest.fn(),
      history: {
        location: {
          search: 'validQuerryParams'
        }
      }
    };
    wrapper = shallow(<Verification {...props} />);
  });
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should check if componentDidMount is called for valid pathname', () => {
    spy = jest.spyOn(props, 'activate');
    wrapper.setProps({
      history: {
        location: {
          search: 'querryParams'
        },
      }
    });
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
});
