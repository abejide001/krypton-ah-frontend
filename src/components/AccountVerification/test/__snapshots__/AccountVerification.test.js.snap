// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountVerification Component should match snapshot 1`] = `
<Fragment>
  <RedirectLoader />
  <Redirect
    push={false}
    to="/"
  />
</Fragment>
`;
